terraform/elasticsearch.tf

```diff
@@ -0,0 +1,40 @@
+resource "aws_elasticsearch_domain" "logging" {
+  domain_name           = "${terraform.workspace}-logging"
+  elasticsearch_version = "7.1"
+  count                 = var.enable_logstash ? 1 : 0
+
+  cluster_config {
+    instance_type = "r5.large.elasticsearch"
+  }
+
+  ebs_options {
+      ebs_enabled = true
+      volume_size = 20
+  }
+
+  snapshot_options {
+    automated_snapshot_start_hour = 23
+  }
+}
+
+resource "aws_elasticsearch_domain_policy" "main" {
+  count       = var.enable_logstash ? 1 : 0
+  domain_name = element(aws_elasticsearch_domain.logging.*.domain_name, count.index)
+
+  access_policies = <<POLICIES
```

Comment:
Can you use an aws_iam_policy_document instead of JSON for this?
https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html